x-pack/plugins/security/server/authentication/can_redirect_request.ts

```diff
@@ -22,12 +23,13 @@ export function canRedirectRequest(request: KibanaRequest) {
   const route = request.route;
   const hasVersionHeader = headers.hasOwnProperty(KIBANA_VERSION_HEADER);
   const hasXsrfHeader = headers.hasOwnProperty(KIBANA_XSRF_HEADER);
+  const hasIternalOriginHeader = headers.hasOwnProperty(X_ELASTIC_INTERNAL_ORIGIN_REQUEST);
 
   const isApiRoute =
     route.options.tags.includes(ROUTE_TAG_API) ||
     route.path.startsWith('/api/') ||
     route.path.startsWith('/internal/');
-  const isAjaxRequest = hasVersionHeader || hasXsrfHeader;
+  const isAjaxRequest = hasVersionHeader || hasXsrfHeader || hasIternalOriginHeader;
```

Comment:
@elastic/kibana-security I took the initiative and applied the implementation. Feel free to suggest changes, I'm not too familiar with your domains.

Reply:
Thanks, @TinaHeiligers. Question for you: Under what conditions will requests be expected to have an `internal origin header`, but _not_ one of the `version` or `xsrf` headers?

Reply:
@legrego good question! I can't think of one TBH. Can you?
Maybe it's better not to add the internal header. I'll make the change.

Reply:
++ yeah I don't think this change is necessary at this point, but thanks for taking this function into consideration!